distribution/test/itests/test-itests-alliance/src/test/java/org/codice/alliance/test/itests/NsiliEndpointTest.java

```diff
@@ -0,0 +1,242 @@
+
+            configureSecurityStsClient();
+
+            ingestedProductId = ingestRecord("alliance.png", "image/png");
+            startHttpListener();
+        } catch (Exception e) {
+            LOGGER.error("Failed in @BeforeExam: ", e);
+            fail("Failed in @BeforeExam: " + e.getMessage());
+        }
+    }
+
+    @AfterExam
+    public void stopHttpListener() {
+        if (server != null) {
+            server.stop();
+        }
+    }
+
+    @Before
+    public void startSampleNsiliClientFeature() throws Exception {
+        getServiceManager().startFeature(true, SAMPLE_NSILI_CLIENT_FEATURE_NAME);
+    }
+
+    @After
+    public void stopSampleNsiliClientFeature() throws Exception {
+        getServiceManager().stopFeature(true, SAMPLE_NSILI_CLIENT_FEATURE_NAME);
+    }
+
+    @Test
+    public void testStandingQueryMgr() throws Exception {
+        SampleNsiliClient sampleNsiliClient = new SampleNsiliClient(Integer.parseInt(RESTITO_STUB_SERVER_PORT.getPort()), NSILI_ENDPOINT_IOR_URL.getUrl());
+
+        sampleNsiliClient.testStandingQueryMgr();
+
+        sampleNsiliClient.cleanup();
+    }
+
+    @Test
+    public void testCatalogMgrGetHitCount() throws Exception {
+        SampleNsiliClient sampleNsiliClient = new SampleNsiliClient(Integer.parseInt(RESTITO_STUB_SERVER_PORT.getPort()), NSILI_ENDPOINT_IOR_URL.getUrl());
+
+        assertThat(sampleNsiliClient.getHitCount(), is(1));
+
+        sampleNsiliClient.cleanup();
+    }
+
+    @Test
+    public void testSubmitQuery() throws Exception {
+        SampleNsiliClient sampleNsiliClient = new SampleNsiliClient(Integer.parseInt(RESTITO_STUB_SERVER_PORT.getPort()), NSILI_ENDPOINT_IOR_URL.getUrl());
+
+        // CatalogMgr
+        DAG[] results = sampleNsiliClient.submit_query();
+        assertThat(results.length, is(1));
+        DAG result = results[0];
+        assertThat(sampleNsiliClient.getProductIdFromDag(result), is(ingestedProductId));
+
+        // ProductMgr
+        DAG parametersDag = sampleNsiliClient.get_parameters(result);
+        assertThat(getAttributeFromDag(parametersDag, NsiliConstants.IDENTIFIER), is(ingestedProductId));
+        assertThat(getAttributeFromDag(parametersDag, NsiliConstants.STATUS), is("NEW"));
+        assertThat(sampleNsiliClient.get_related_file_types(result), is(new String[]{NsiliConstants.THUMBNAIL_TYPE}));
+        final String expectedThumbnailFilename = ingestedProductId + "-THUMBNAIL.jpg";
+        assertThat(sampleNsiliClient.get_related_files(result), is(new String[]{expectedThumbnailFilename}));
+        assertStubServerPutRequest(expectedThumbnailFilename, "image/jpeg");
+
+        // OrderMgr
+        String orderResultFilename = sampleNsiliClient.testOrder(result);
+        // TODO test processing packageElements of the order response https://codice.atlassian.net/browse/CAL-192
+        assertStubServerPutRequest(orderResultFilename, "application/x-tar");
+
+        deleteMetacard(ingestedProductId);
```

Comment:
`deleteMetacard` should be called in the `@After` method to make sure it always gets deleted.